### PR TITLE
ss-279: Add API to delete user

### PR DESCRIPTION
**Pull request by anton-otroshchenko**

### CODE REVIEW

**Comment by anton-otroshchenko** [comment]
image

Pass only token

**Review by Holovashcenko — APPROVED** [review]

File: backend/src/bundles/users/user.service.ts [hunk 1]
```diff
@@ -63,8 +63,8 @@ class UserService implements IService {
         return updatedUser.toObject();
     }
 
-    public delete(): ReturnType<IService['delete']> {
-        return Promise.resolve(true);
+    public async deleteUser(id: string): Promise<UserEntity | undefined> {
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/user.service.ts:66]: I not sure that we should return `UserEntity` on delete

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/users/user.service.ts (hunk 1 above)
Comment (anton-otroshchenko) [reply backend/src/bundles/users/user.service.ts]: fixed

**Review by Evcat6 — APPROVED** [review]

File: shared/src/enums/exception-message.enum.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ enum ExceptionMessage {
     INVALID_CREDENTIALS = 'You entered an incorrect e-mail address or password',
     EMAIL_ALREADY_EXISTS = 'User with this email already exists',
     INVALID_TOKEN = 'Invalid token',
-    AUTHORIZE_ERROR = 'authorize error'
+    AUTHORIZE_ERROR = 'authorize error',
```
Comment (nikita-remeslov) [inline shared/src/enums/exception-message.enum.ts:5]:
```suggestion
    AUTHORIZE_ERROR = 'You should authorize',
```

File: backend/src/bundles/users/user.service.ts [hunk 3]
```diff
@@ -63,8 +63,9 @@ class UserService implements IService {
         return updatedUser.toObject();
     }
 
-    public delete(): ReturnType<IService['delete']> {
-        return Promise.resolve(true);
+    public async deleteUser(id: string): Promise<boolean | undefined> {
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/user.service.ts:66]:
```suggestion
    public async deleteUser(id: string): Promise<boolean> {
```

File: backend/src/bundles/users/user.repository.ts [hunk 4]
```diff
@@ -77,8 +77,14 @@ class UserRepository implements Omit<IRepository, 'update' | 'delete'> {
         return UserEntity.initialize(user);
     }
 
-    public delete(): ReturnType<IRepository['delete']> {
-        return Promise.resolve(true);
+    public async deleteUser(id: string): Promise<UserEntity | undefined> {
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/user.repository.ts:80]: return boolean value

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/users/user.service.ts (hunk 3 above)
Comment (anton-otroshchenko) [reply backend/src/bundles/users/user.service.ts]: fixed

File: shared/src/enums/exception-message.enum.ts (hunk 2 above)
Comment (anton-otroshchenko) [reply shared/src/enums/exception-message.enum.ts]: fixed

**Review by VItaliy-Kalenichenko — APPROVED** [review]

**Comment by VItaliy-Kalenichenko** [comment]
Resolve conflicts

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**